src/domain/transaction/mod.rs

```diff
@@ -404,19 +407,25 @@ impl RelayerTransactionFactory {
                     .map_err(|e| TransactionError::NetworkConfiguration(e.to_string()))?;
 
                 let signer_service = EvmSignerFactory::create_evm_signer(&signer)?;
-                let price_calculator =
-                    PriceCalculator::new(EvmGasPriceService::new(evm_provider.clone(), network));
+                let network_extra_fee_calculator =
+                    get_network_extra_fee_calculator_service(network, evm_provider.clone());
+                let price_calculator = PriceCalculator::new(
+                    EvmGasPriceService::new(evm_provider.clone(), network),
+                    Some(network_extra_fee_calculator),
```

Comment:
Why do we use Option(Some) here? Because of tests?

I see that we would always have instance of this service so we might want to remove Option usage.

In implementation we are returning None and we have mock type so I think we should be able to remove Option usage.